pkg/test/cel/grpcroute_test.go

```diff
@@ -297,6 +298,118 @@ func TestGRPCRouteRule(t *testing.T) {
 	}
 }
 
+func TestGRPCMethodMatch(t *testing.T) {
+	service := "foo.Test.Example"
+	method := "Login"
+	regex := ".*"
+
+	tests := []struct {
+		name       string
+		method     gatewayv1a2.GRPCMethodMatch
+		wantErrors []string
+	}{
+		{
+			name: "valid GRPCRoute with 1 service in GRPCMethodMatch field",
+			method: gatewayv1a2.GRPCMethodMatch{
+				Service: &service,
+			},
+		},
+		{
+			name: "valid GRPCRoute with 1 method in GRPCMethodMatch field",
+			method: gatewayv1a2.GRPCMethodMatch{
+				Method: &method,
+			},
+		},
+		{
+			name: "invalid GRPCRoute missing service or method in GRPCMethodMatch field",
+			method: gatewayv1a2.GRPCMethodMatch{
+				Service: nil,
+				Method:  nil,
+			},
+			wantErrors: []string{"One or both of 'service' or 'method"},
+		},
+		{
+			name: "GRPCRoute use regex in service and method with undefined match type",
```

Comment:
Nit: recommend replacing "use" with "uses" in most test case names below this point
```suggestion
			name: "GRPCRoute uses regex in service and method with undefined match type",
```

Reply:
Yes definitely. Done